src/services/refactors/convertParamsToDestructuredObject.ts

```diff
@@ -488,11 +512,12 @@ namespace ts.refactor.convertParamsToDestructuredObject {
         return getTextOfIdentifierOrLiteral(paramDeclaration.name);
     }
 
-    function getClassNames(constructorDeclaration: ValidConstructor): Identifier[] {
+    function getClassNames(constructorDeclaration: ValidConstructor): (Identifier | Modifier)[] {
         switch (constructorDeclaration.parent.kind) {
             case SyntaxKind.ClassDeclaration:
                 const classDeclaration = constructorDeclaration.parent;
-                return [classDeclaration.name];
+                if (classDeclaration.name) return [classDeclaration.name];
+                return [findModifier(classDeclaration, SyntaxKind.DefaultKeyword)!];

```

Comment:
>! [](start = 81, length = 1)

We expect a `default` modifier because we wouldn't otherwise have called this helper?

Reply:
does findModifier assert if the desired kind isn't present? it would be nice to have an assertion here.

Reply:
we expect a `default` modifier because when validating the function in `isValidFunctionDeclaration`, if a function declaration or a constructor's class declaration has no name, it has to have the default modifier so we can search for it. this is checked in the `hasNameOrDefault` function.
i have added assertions and comments about this.